services/userService.js

```diff
@@ -1,57 +1,60 @@
-
-const bcrypt = require('bcrypt')
-const { createToken } = require('../utils/jwt');
-const {userTokenName} = require('../config/app-config');
-
-
+const bcrypt = require("bcrypt");
+const { createToken } = require("../utils/jwt");
+const { userTokenName } = require("../config/app-config");
 
 const UserTypes = {
-  doctor: require("../models/doctorsModel"),
+  doctor: require("../models/doctorModel"),
   pharmacist: require("../models/pharmacistModel"),
-  patient: require("../models/patitentModel"),
+  patient: require("../models/patientModel"),
 };
 
+const register = async (
+  firstName,
+  lastName,
+  username,
+  email,
+  password,
+  role,
+  profileInfo
+) => {
+  const model = UserTypes[role];
+  const existing = await model
+    .findOne({ email })
+    .collation({ locale: "en", strength: 2 });
+  if (existing) {
+    throw new Error("Email is already exist");
```

Comment:
The correct message here should be "Email is already in use" or something similar